### PR TITLE
feat(*): add helm rollback functionality

### LINKED ISSUE #1004: Feature: `helm rollback RELEASE VERSION` can rollback to a previous version
A command named `helm rollback RELEASE [VERSION]` should be added to allow a user to replace the current version of a release with a previous version of a release.

Ideally, this would work not as an "upgrade" to an old version, but as a hard reset to the old version. In other words, it would re-create the old resources as they were, destroying the newer version first.

In the backend, the versioning and storage should look like this:

Record   |  Description
------- | -----
foo, v1    | Initial release

User runs `helm update foo`

Record   |  Description
------- | -----
foo, v1    | Initial release
foo, v2   | Upgraded release

User runs `helm rollback foo v1`

Record   |  Description
------- | -----
foo, v1    | Initial release
foo, v2   | Upgraded release
foo, v3 | Copy of "Initial release"

This way, a pristine and linear history is maintained.

Open question: For a rollback, to we re-template the release or re-release exactly? Re-templating might trigger certain functions like `date` to generate new values. To me, this sounds like a good thing.

### PR DESCRIPTION
**Pull request by michelleN**
This feature allows you to rollback release to the
previous version of release.
resolves #1004

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/helm/1218)

### CODE REVIEW

File: cmd/tiller/release_server.go [hunk 1]
```diff
@@ -349,6 +346,114 @@ func (s *releaseServer) prepareUpdate(req *services.UpdateReleaseRequest) (*rele
...
+	if err != nil {
+		return nil, err
+	}
+
+	rel, err := s.performRollback(currentRelease, targetRelease, req)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := s.env.Releases.Create(targetRelease); err != nil {
```
Comment (fibonacci1729) [inline cmd/tiller/release_server.go]: @adamreese pointed out that if dry-rolling, this storage entry is still created causing dissonance between the reality and perception of history. Simple solution seems to be to just move this line to the end of performRollback.

**Comment by adamreese** [comment]
googlebot, same team man

**Comment by adamreese** [comment]
@googlebot confirm

**Comment by adamreese** [comment]
@googlebot ok

**Comment by adamreese** [comment]
TODO: accept a release version on the cli
ref #1244

**Pull request merged**